tests/nested/manual/uc-update-command-line-secure/task.yaml

```
@@ -0,0 +1,30 @@
+summary: Update command line from gadget in secure boot
+
+systems: [ubuntu-20.04-64, ubuntu-22.04-64]
+
+environment:
+  NESTED_ENABLE_SECURE_BOOT: true
+
+prepare: |
+  CHANNEL="$(tests.nested show version)/stable"
+  snap download --basename=pc --channel="${CHANNEL}" pc
+  unsquashfs -d pc pc.snap
+  KEY_NAME=$(tests.nested download snakeoil-key)
+  SNAKEOIL_KEY="$PWD/$KEY_NAME.key"
+  SNAKEOIL_CERT="$PWD/$KEY_NAME.pem"
+  tests.nested secboot-sign file pc/shim.efi.signed "$SNAKEOIL_KEY" "$SNAKEOIL_CERT"
+  tests.nested secboot-sign file pc/grubx64.efi "$SNAKEOIL_KEY" "$SNAKEOIL_CERT"
+  snap pack pc "$(tests.nested get extra-snaps-path)"
+
+  echo "some_useless_parameter" >>pc/cmdline.extra
+  snap pack pc --filename=pc_2.snap
+
+  tests.nested build-image core
+  tests.nested create-vm core
+
+execute: |
+  tests.nested copy pc_2.snap
+  boot_id="$(tests.nested boot-id)"
+  tests.nested exec "sudo snap install --dangerous pc_2.snap"
+  tests.nested wait-for reboot "${boot_id}"
+  tests.nested exec "cat /proc/cmdline" | MATCH ".*some_useless_parameter.*"
```

you can use MATCH in the nested VM 
remote.exec 'MATCH ".*some_useless_parameter.*" < /proc/cmdline'

those tools are deployed to the nested VM after it is created. 
```suggestion
  tests.nested exec "cat /proc/cmdline" | MATCH ".*some_useless_parameter.*"
```

---

Done.